src/EnergyPlus/WaterCoils.cc

```diff
@@ -1927,7 +1927,7 @@ namespace WaterCoils {
 				bPRINT = false; // do not print this sizing request
 				TempSize = AutoSize; // get the autosized air volume flow rate for use in other calculations
 				SizingString.clear(); // doesn't matter
-				CompType = cAllCoilTypes(16); // "Coil:Heating:Water"
+				CompType = cAllCoilTypes(Coil_HeatingWater); // "Coil:Heating:Water"
 				CompName = WaterCoil( CoilNum ).Name;
 				RequestSizing( CompType, CompName, HeatingAirflowSizing, SizingString, TempSize, bPRINT, RoutineName );
 				// reset the design air volume flow rate for air loop coils only
```

Comment:
These hard-coded coil types caused problems when new coil types were added to DataHVACGlobals.